docs/generate/readme.md

```diff
@@ -0,0 +1,355 @@
+|`--output-folder=DIRECTORY`|The location for generated files. If not specified, uses `./Generated` as the default|
+|`--clear-output-folder`|Clear all contents from our output folder before outputting your newly generated code into that folder|
+|`--namespace=NAMESPACE`|sets the namespace to use for the generated code|
+|`--add-credential`|If specified, the generated client will require a credential to make network calls. See [TODO] for information on how to authenticate to our generated clients|
+|`--tag=VALUE`|Preferred way to have conditional configurations. I.e., in my configuration file, I can set the `input-file` equal to different values depending on the `VALUE` passed through the `tag` flag. See our [Adding Tags When Generating](#adding-tags-when-generating) section for more information|
+
+## Most Basic: Generating with a Single File on the Command Line
+
+The first step in an AutoRest journey usually starts with generating a single OpenAPI file. We will also show
+how to set options during generation by building up our command-line step-by-step.
+
+The first step here is to have your OpenAPI file definition of your client ([docs](./openapi/introduction.md)) on how to do that).
+This example will use an example OpenAPI definition found [here](../../openapi/examples/pets.json), so feel free to follow along with
+our sample code. To get things started, the command that starts AutoRest on the command line is `autorest`, so this is what our command line
+looks like to start with
+
+```
+autorest
+```
+
+Next, we want to tell AutoRest which swagger file to generate. We do this by passing our swagger file through the `--input-file` flag, see [common
+flags](#common-flags) for a description of its uses. Adding this to our command, we have
+
+```
+autorest --input-file=pets.json
+```
+
+We also need to tell AutoRest what language we want our SDK to be in, which we specify using our [language flags](#language-flags). For the sake of this example,
+let's say we want to generate Python code. Adding this to our command line, we get
+
+```
+autorest --input=file=pets.json --python
+```
+
+In our final required step, we need to tell AutoRest where to output the generated SDK. We do this through the `--output-folder` flag (once again, see [common
+flags](#common-flags) for more information). Putting this all together, we have:
+
+```
+autorest --input-file=pets.json --python --output-folder=generated/
+```
+
+There are many other flags you can specify when generating. As an add-on, let's say we want to generate our code under the namespace `pets`. This gives us:
+
+```
+autorest --input-file=pets.json --python --output-folder=generated/ --namespace=pets
+```
+And this concludes our basic example of generating with AutoRest. Continue reading to the next section to see our recommend way of generating AutoRest.
+
+## Slighly More Complicated, But Preferred: Generating with a Configuration File
```

Comment:
Let's be more positive: this is about people avoiding to have to save huge gigantic script with lines of CLI options, the Readme is a way to save those options (but syntax is autorest specific).

We could do as title (to stay in the same mood of sentence)
- Save CLI args into options into a configuration file: the preferred option
- Keep your options in one place: the configuration file

I would avoid the word "complicated" that put the mindset in warning mode even before we started to read :p

Reply:
sounds good!